flaky test TestGeneralProperties

https://do.pingcap.net/jenkins/blue/rest/organizations/jenkins/pipelines/pingcap/pipelines/tidb/pipelines/ghpr_unit_test/runs/32777/nodes/59/steps/65/log/?start=0

```
    split_test.go:34: seed: 1694425751
```

```
    split_test.go:80: 
        	Error Trace:	br/pkg/lightning/backend/external/split_test.go:80
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestGeneralProperties
        	Messages:   	splitKeys: [[226] [226 25 184 61 106 108 211 54 94 85 60 229 12 30 118 45 58 192 96 246 8 143 255 143 75 85 185] [226 39 231 179 198 138] [226 78 191 160 137 244 169 125 4 42 125 8 81 152 251 188 88 154 24 96 111 139 122 210 48 243 21 31 251 189 18 113 102 221 184 133 123 44 143 230 88 217 254 122 25 19 228 104 39 203 1 117 179 125 203 171 201 168 77 244 13 120 10 253 85 235 87 157 251 58 224 106 72 28 34 215 117 41 244 248 245 3 188 111 65 105 184 217 252 201 114 228 84 204 43 57 113 244 88] [226 80 50 143 245 63 26 214 24 242 193 134 139 183 63 217 220 107 167 194 32 163 226 14 16 209 16 50 191 188 81 255 59 56 41 32 239 98 132 253 98 213 214 126 105 53 124 61 22 85 77 49 252 129 172 140 17 101 145 106 8 162 195 226 168 3 20 181 249 140 10 233 134 107 245 130 194 143 199 7 153 206 243 133 169 52 30 84 113 111 140 219 61 24 11 235] [226 119 162 198 216 31 180 29 182 196 101 18 32 210 182 91 62 82 106 126 45 71 91 200 248 36 41 34 72 221 237 9 89 170 137 236 171 72 95 50 235 140 115 183] [226 162 206 246 163 47 165 193 192 234 43 220 108 189 112 7 61 59 46 38 22 41 73 104 229] [227 19 104 42 239 107 254 244 186 89 54 143 251 223 32 35 9 132 57 13 137 40 218 40 55 45 180 168 19 190 163 91 195 115 119 50 154 145 4 147 122 58 202 167 158 72 19 88 229 179 5 242 28 202 238 13 5 231 190 136 16 207 199 118 247 3 61 125 48 20 213 63 201 11 38 250 172 242 253 58 10 165 199 123 23 242 237 228 16 230 176 126 236 79] [227 187 63 154 237 66 1 126 190 55 209 235 37 220 190 114 175 159 153 102 51 167 224 121 156 85 14 44 113 221 10 228 157 72 115 127 81 42 123 23 10 166 196 177 15 230 3 243 122 86 83 154 193 93 219 176 126 154 178 223 221 255 16 25 205]], lastEndKey: [226]
--- FAIL: TestGeneralProperties (0.07s)
```

--- a/br/pkg/lightning/backend/external/split_test.go
+++ b/br/pkg/lightning/backend/external/split_test.go
@@ -42,10 +42,14 @@ func TestGeneralProperties(t *testing.T) {
 	for i := range keys {
 		keyLen := rand.Intn(100) + 1
 		valueLen := rand.Intn(100) + 1
-		keys[i] = make([]byte, keyLen)
-		values[i] = make([]byte, valueLen)
-		rand.Read(keys[i])
-		rand.Read(values[i])
+		keys[i] = make([]byte, keyLen+2)
+		values[i] = make([]byte, valueLen+2)
+		rand.Read(keys[i][:keyLen])
+		rand.Read(values[i][:valueLen])
+		keys[i][keyLen] = byte(i / 255)
+		keys[i][keyLen+1] = byte(i % 255)
+		values[i][valueLen] = byte(i / 255)
+		values[i][valueLen+1] = byte(i % 255)
 	}
 
 	dataFiles, statFiles, err := MockExternalEngine(memStore, keys, values)

